test/request_source/request_source_plugin_test.cc

```diff
@@ -223,6 +223,63 @@ TEST_F(FileBasedRequestSourcePluginTest,
   EXPECT_EQ(header3->getPathValue(), "/a");
 }
 
+TEST_F(
+    FileBasedRequestSourcePluginTest,
+    CreateRequestSourcePluginMultipleTimesWithDifferentConfigsCreatesDifferentWorkingRequestsSources) {
+  nighthawk::request_source::FileBasedOptionsListRequestSourceConfig config_ab =
+      MakeFileBasedPluginConfigWithTestYaml(Nighthawk::TestEnvironment::runfilesPath(
+          "test/request_source/test_data/test-config-ab.yaml"));
+  nighthawk::request_source::FileBasedOptionsListRequestSourceConfig config_c =
+      MakeFileBasedPluginConfigWithTestYaml(Nighthawk::TestEnvironment::runfilesPath(
+          "test/request_source/test_data/test-config-c.yaml"));
+  config_ab.set_num_requests(2);
+  config_c.set_num_requests(2);
+
+  Envoy::ProtobufWkt::Any config_ab_any;
+  config_ab_any.PackFrom(config_ab);
+  Envoy::ProtobufWkt::Any config_c_any;
+  config_c_any.PackFrom(config_c);
+
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<RequestSourcePluginConfigFactory>(
+          "nighthawk.file-based-request-source-plugin");
+  auto header_ab = Envoy::Http::RequestHeaderMapImpl::create();
```

Comment:
Can we use the actual std::unique_ptr<TerminationPredicate> type here and below?